issue 7 resolved

C4mperpr0 opened the pull request:
This seemed to only apply for one class (BatLevel). BatLevel has now from_percentage method.

C4mperpr0 committed:
issue 7 resolved

This seemed to only apply for one class (BatLevel). BatLevel has now from_percentage method.

olimaye commented on philander/gasgauge.py:
```diff
@@ -209,19 +208,8 @@ def rateSOC( self, soc ):
-            ret = BatLevel.empty
-        else:
-            ret = BatLevel.deepDischarge
+        # TODO: this function is now redundant. Should it be removed and BatLevel.from_percentage be called directly?
```
Right. rateSOC() shall be removed from the Gas Gauge interface.
Either do it right now, as we still are in an early stage of the implementation. Or file an issue to have it done, later, please.

olimaye commented on philander/stc311x.py:
```diff
@@ -257,18 +257,8 @@ def rateSOC(self, soc):
-            ret = BatLevel.empty
-        else:
-            ret = BatLevel.deepDischarge
+        # TODO: why is here not this function in the superclass used and why is it not static there?
```
Obsolete comment.
Further, think of removing the whole function.

olimaye commented on philander/stc311x.py:
```diff
@@ -257,18 +257,8 @@ def rateSOC(self, soc):
+        ret = BatLevel.from_percentage(soc)
         return ret
 
     def getRatedSOC(self):
```
remove. Let the superclass do this job.

olimaye commented on philander/stc311x.py:
```diff
@@ -287,6 +278,7 @@ def getRatedSOCStr(self):
         :return: The next-lower battery level corresponding to the current SOC.
         :rtype: String
         """
+        # TODO: why is here not this function in the superclass used and why is it not static there?
```
Right. Remove this function and let the superclass do this job.

olimaye reviewed (changes requested):
Nice idea. I think we can improve.

olimaye commented:
We also have to clarify the naming convention. Up to now, CamelCaps was mainly used, which is also my preference. Have to point this out in related documentation.

C4mperpr0 committed:
improved logic

from_percentage will now return Percentage.invalid, if percentage is smaller than 0 (no matter the actual value of Percentage.invalid).

C4mperpr0 committed:
fixed logic again

reworked logic again to have Level.invalid as intial return value but independent of its actual value

C4mperpr0 committed:
renamed function, fixed small mistake

olimaye reviewed (approved):
Ok now.
